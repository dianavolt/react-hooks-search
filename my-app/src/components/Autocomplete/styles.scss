.input-suggestions {
    padding: 10px;
    border: 1px solid grey;
    border-radius: 2px;
    margin-top: 2px;
    width: 100%;
    width:200px;
    right: 0;
    left: 0;
    box-sizing: border-box;
    background: #ffffff;
    text-align: left;
    z-index: 2;

    .search-link {
        display: block;
        margin-top: 10px;
        padding: 4px 0;
    }
}