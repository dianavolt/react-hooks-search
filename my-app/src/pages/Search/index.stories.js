import React from 'react'

import SearchPage from '.'

export default {
  title: 'Pages/SearchPage',
  component: SearchPage,
}

const Template = (args) => <SearchPage {...args} />

export const Primary = Template.bind({})
Primary.args = {
location: {
  search: '?query=testing'
  }
}

export const EmptyQuery = Template.bind({})
Primary.args = {
location: {
  search: '?query='
  }
}
