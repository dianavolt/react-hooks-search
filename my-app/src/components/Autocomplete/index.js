import React from 'react'
import ReactAutocomplete from 'react-autocomplete'
import PropTypes from 'prop-types'

import Input from '../Input'

import './styles.scss'

const Autocomplete = ({ articles, searchVal, onSearchChange }) => (
    <ReactAutocomplete
        items={articles}
        renderInput={Input}
        inputProps={{placeholder: 'Enter search term...'}}
        getItemValue={item => item.label}
        renderMenu={(children, value, style) => {
            return (articles && articles.length > 0)?
                (<div style={{...style}} className="input-suggestions">
                    {children}
                    <a href={`/search?query=${value}`} className='search-link'>
                        See all results
                    </a>
                </div>
                ) : <></>
        }}
        renderItem={(item, highlighted) =>
            <div
                key={item.id}
                style={{ backgroundColor: highlighted ? '#eee' : 'transparent'}}
            >
                <a href={item.id} target="_blank">{item.label}</a>
            </div>
        }
        value={searchVal}
        onChange={onSearchChange}
    />
)

Autocomplete.propTypes = {
    articles: PropTypes.array,
    searchVal:PropTypes.string
}

export default Autocomplete
