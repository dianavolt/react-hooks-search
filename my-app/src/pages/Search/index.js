import React from 'react'
import Container from '../../components/Container'
import Autocomplete from '../../components/Autocomplete'
import { useSearch } from '../../hooks'

const Search = ({...props}) => {
    const search = props.location.search
    const params = new URLSearchParams(search)
    const query = params.get('query')

    const { articles, status } = useSearch(query, 50)

    return (
        <div>
            <Container>
                {({ articles, searchVal, onSearchChange }) => 
                <Autocomplete 
                    articles={articles}
                    searchVal={searchVal}
                    onSearchChange={onSearchChange}
                />
                }
            </Container>
            {!articles.length && status === 'SUCCESS'
                ? <h3>No items found for: {query}</h3>
                : articles.map(article => {
                    return (
                        <div key={article.id}>
                            <a href={article.id} target="_blank">
                                {article.label}
                            </a>
                        </div>
                    )
                })
            }
        </div>
    )
}

export default Search