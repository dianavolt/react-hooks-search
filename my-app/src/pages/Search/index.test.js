import React from 'react'
import { mount, shallow } from 'enzyme'

import Search from '.'
import Autocomplete from '../../components/Autocomplete'
import SearchItems from '../../components/SearchItem'

const render = (props, children) => mount(<Search {...props}>{children}</Search>)

describe('Search component', () => {
    let test
    let props
    let children = ({searchVal, onSearchChange, articles}) => <Autocomplete searchVal={searchVal}
            onSearchChange={onSearchChange}
            articles={articles}/>

    describe('with props', () => {
        beforeEach(() => {
            props = {
                location: {
                    search: 'test search'
                }
            }
            test = render(props, children)
        })

        it('should match snapshot', () => {
            expect(test).toMatchSnapshot()
        })
    })

    describe('when SearchItem is called', () => {
        const articles = [
            {
                id: '3475987',
                label: 'FIELD1'
            }
        ]

        it('should match snapshot', () => {
            expect(articles.map(article => shallow(<SearchItems {...props} key={article.id}/>))).toMatchSnapshot()
        })
        
        it('should render article id', () => {
            expect(articles.map(article => article.id)).toEqual(['3475987'])
        })
    })
})
