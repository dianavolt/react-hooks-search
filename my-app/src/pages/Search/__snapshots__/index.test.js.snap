// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component when SearchItem is called should match snapshot 1`] = `
Array [
  <div
    className="search-item"
  >
    <a
      target="_blank"
    />
    <p />
  </div>,
]
`;

exports[`Search component with props should match snapshot 1`] = `
<Search
  location={
    Object {
      "search": "test search",
    }
  }
>
  <div>
    <Container>
      <Autocomplete
        articles={Array []}
        onSearchChange={[Function]}
        searchVal=""
      >
        <Autocomplete
          autoHighlight={true}
          getItemValue={[Function]}
          inputProps={
            Object {
              "placeholder": "Enter search term...",
            }
          }
          isItemSelectable={[Function]}
          items={Array []}
          menuStyle={
            Object {
              "background": "rgba(255, 255, 255, 0.9)",
              "borderRadius": "3px",
              "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
              "fontSize": "90%",
              "maxHeight": "50%",
              "overflow": "auto",
              "padding": "2px 0",
              "position": "fixed",
            }
          }
          onChange={[Function]}
          onMenuVisibilityChange={[Function]}
          onSelect={[Function]}
          renderInput={[Function]}
          renderItem={[Function]}
          renderMenu={[Function]}
          selectOnBlur={false}
          value=""
          wrapperProps={Object {}}
          wrapperStyle={
            Object {
              "display": "inline-block",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              className="input-field"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter search term..."
              role="combobox"
              value=""
            />
          </div>
        </Autocomplete>
      </Autocomplete>
    </Container>
  </div>
</Search>
`;
